src/gtc/dace/transformations.py

```diff
@@ -0,0 +1,116 @@
+import dace
+from dace.transformation.dataflow import TrivialMapElimination
+from dace.transformation.helpers import nest_state_subgraph
+from dace.transformation.interstate import InlineTransients
+
+
+def eliminate_trivial_maps(sdfg: dace.SDFG):
+    """Remove maps and map ranges where the iteration is over a single index."""
+    applied = True
+    while applied:
+        applied = False
+        for map_entry, state in sdfg.all_nodes_recursive():
+            if isinstance(map_entry, dace.nodes.MapEntry):
+                if map_entry.map.schedule in {
+                    dace.ScheduleType.Sequential,
+                    dace.ScheduleType.CPU_Multicore,
+                }:
+                    # exclude maps with empty edges as workaround for a bug in TrivialMapElimination
+                    if any(
+                        edge.data.data is None
+                        for edge in state.in_edges(map_entry)
+                        + state.out_edges(state.exit_node(map_entry))
+                    ):
+                        continue
+                    try:
+                        TrivialMapElimination.apply_to(
+                            state.parent, map_entry=map_entry, verify=True
+                        )
+                        applied = True
+                        break
+                    except ValueError:
+                        continue
+
+
+class InlineThreadLocalTransients(dace.transformation.SingleStateTransformation):
+    """
+    Inline and tile thread-local transients.
+
+    Inlines transients like `dace.transformations.interstate.InlineTransients`, however only applies to OpenMP map
+    scopes but also makes the resulting local arrays persistent and thread-local. This reproduces `cpu_kfirst`-style
+    transient tiling.
+    """
+
+    map_entry = dace.transformation.transformation.PatternNode(dace.nodes.MapEntry)
+
+    @classmethod
+    def expressions(cls):
+        return [dace.sdfg.utils.node_path_graph(cls.map_entry)]
+
+    def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
+        map_entry = self.map_entry
+
+        if not map_entry.schedule == dace.ScheduleType.CPU_Multicore:
+            return False
+
+        scope_subgraph = graph.scope_subgraph(map_entry, include_entry=False, include_exit=False)
+        if len(scope_subgraph) > 1 or not isinstance(
+            scope_subgraph.nodes()[0], dace.nodes.NestedSDFG
+        ):
+            return False
+
+        toremove = InlineTransients._candidates(sdfg, graph, scope_subgraph.nodes()[0])
+        return len(toremove) > 0
+
+    def apply(self, graph, sdfg):
+        map_entry = self.map_entry
+
+        scope_subgraph = graph.scope_subgraph(map_entry, include_entry=False, include_exit=False)
+        nsdfg_node = scope_subgraph.nodes()[0]
+        toremove = InlineTransients._candidates(sdfg, graph, nsdfg_node)
```

Comment:
misleading return value name

Reply:
I put `candidates` now and check if it has actually been applied in `apply`.